lib/ddtrace/contrib/sinatra/header_tagger_middleware.rb

```diff
@@ -0,0 +1,72 @@
+require 'ddtrace/contrib/sinatra/request_span'
+
+module Datadog
+  module Contrib
+    module Sinatra
+      # Middleware used for automatically tagging configured headers and handle request span
+      class HeaderTaggerMiddleware
+        def initialize(app)
+          @app = app
+        end
+
+        def call(env)
+          span = RequestSpan.span!(env)
+
+          # Request headers
+          parse_request_headers(env).each do |name, value|
```

Comment:
So if I understand correctly, you mentioned that the reason for doing this header tagging in the middleware, instead of in the existing tracer extension, was because we can only access some data we need from this middleware layer. Is that right?

If that is correct, what data can we exclusively access from this middleware, that we can't access in the tracer?